api/types/matchers_aws.go

```diff
@@ -274,3 +278,35 @@ func (m *AWSMatcher) CheckAndSetDefaults() error {
 	}
 	return nil
 }
+
+// HasOrganizationMatcher returns true if the matcher has an organization ID set.
+func (m *AWSMatcher) HasOrganizationMatcher() bool {
+	return m.Organization != nil && m.Organization.OrganizationID != ""
+}
+
+func (m *AWSMatcher) validateOrganizationAccountDiscovery() error {
+	if m.Organization == nil {
+		return nil
+	}
+
+	if m.Organization.OrganizationID == "" {
+		// If no organization is set, accounts filter cannot be set.
+		if m.Organization.OrganizationalUnits != nil {
+			if len(m.Organization.OrganizationalUnits.Exclude) > 0 || len(m.Organization.OrganizationalUnits.Include) > 0 {
+				return trace.BadParameter("organization id is required when using accounts filter")
+			}
+		}
+
+		return nil
+	}
+
+	if m.AssumeRole == nil || m.AssumeRole.RoleName == "" {
```

Comment:
I think we should also enforce that role arn is not set

Reply:
Oops, totally missed it. Sorry for the noise.